moped-editor/src/utils/mapHelpers.js

```diff
@@ -828,3 +969,47 @@ export const layerSelectStyles = {
 export const drawnLayerNames = Object.entries(mapConfig.layerConfigs)
   .filter(([layerName, layer]) => layer.layerDrawn)
   .map(([layerName, layer]) => layer.layerIdName);
+
+/**
+ * Reconstructs a GeoJSON collection and renames all its features using
+ * a provided layer name id (as specified in the configuration settings).
+ * @param {Object} projectOtherFeaturesCollection - A GeoJSON collection
+ * @param {Object} newLayerNameConfig - An object containing a key value pair of names based on feature type
+ */
+export const useTransformProjectFeatures = (
+  projectOtherFeaturesCollection,
+  newLayerNameConfig // We need to generate a new collection
+) => ({
+  // First, enter the type (which never changes)
+  type: "FeatureCollection",
+  // Then, create the features attribute with the output of a map
+  features: (projectOtherFeaturesCollection?.features ?? []).map(feature => ({
+    // For every feature, first copy the element
+    ...feature,
+    // Then, overwrite the feature's 'properties' attribute
+    properties: {
+      // With a copy of the existing feature's properties
+      ...feature.properties,
+      // And with an overwritten sourceLayer attribute (if it can find it)
+      sourceLayer:
+        // Try to find it in the configuration object
+        newLayerNameConfig[feature.geometry.type] ??
+        // or default to the original one if not found in the config
+        feature.properties.sourceLayer,
+    },
+  })),
+});
+
+/**
+ * Instantiates a saveActionState and a dispatch function
+ * @return {Object}
+ */
+export const useSaveActionReducer = () => {
+  const [saveActionState, saveActionDispatch] = useReducer(
+    mapSaveActionReducer, // The reducer function handler
+    null, // Initializer argument
+    mapSaveActionInitialState // Initial state
+  );
+
+  return { saveActionState, saveActionDispatch };
+};
```

Comment:
This is a custom hook that returns the the reducer state, and the reducer dispatch function we use to track events within the component editor.